src/models/arellano_vfi.jl

```diff
@@ -0,0 +1,261 @@
+    ygrid = exp(ly)
+    ydefgrid = min(.969 * mean(ygrid), ygrid)
+
+    # Define value functions (Notice ordered different than Python to take
+    # advantage of column major layout of Julia)
+    vf = zeros(nB, ny)
+    vd = zeros(1, ny)
+    vc = zeros(nB, ny)
+    policy = Array(Int, nB, ny)
+    q = ones(nB, ny) .* (1 / (1 + r))
+    defprob = Array(Float64, nB, ny)
+
+    return Arellano_Economy(β, γ, r, ρ, η, θ, ny, nB, ygrid, ydefgrid, Bgrid, Π,
+                            vf, vd, vc, policy, q, defprob)
+end
+
+u(ae::Arellano_Economy, c) = c^(1 - ae.γ) / (1 - ae.γ)
+_unpack(ae::Arellano_Economy) =
+    ae.β, ae.γ, ae.r, ae.ρ, ae.η, ae.θ, ae.ny, ae.nB
+_unpackgrids(ae::Arellano_Economy) =
+    ae.ygrid, ae.ydefgrid, ae.Bgrid, ae.Π, ae.vf, ae.vd, ae.vc, ae.policy, ae.q, ae.defprob
+
+
+# ------------------------------------------------------------------- #
+# Write the value function iteration
+# ------------------------------------------------------------------- #
+#=
+This function performs the one step update -- Using current value
+functions and their expected value, it updates the value function
+at every state by solving for the optimal choice of savings
+
+Note: Updates value functions in place.
+=#
+function one_step_update!(ae::Arellano_Economy, EV, EVd, EVc)
+
+    # Unpack stuff
+    β, γ, r, ρ, η, θ, ny, nB = _unpack(ae)
+    ygrid, ydefgrid, Bgrid, Π, vf, vd, vc, policy, q, defprob = _unpackgrids(ae)
+    zero_ind = searchsortedfirst(Bgrid, 0.)
+
+    for iy=1:ny
+        y = ae.ygrid[iy]
+        ydef = ae.ydefgrid[iy]
+
+        # Value of being in default with income y
+        defval = u(ae, ydef) + β*(θ*EVc[zero_ind, iy] + (1-θ)*EVd[1, iy])
+        ae.vd[1, iy] = defval
+
+        for ib=1:nB
+            B = ae.Bgrid[ib]
+
+            current_max = -1e14
+            pol_ind = 0
+            for ib_next=1:nB
+                c = max(y - ae.q[ib_next, iy]*Bgrid[ib_next] + B, 1e-14)
+                m = u(ae, c) + β * EV[ib_next, iy]
+
+                if m > current_max
+                    current_max = m
+                    pol_ind = ib_next
+                end
+
+            end
+
+            # Update value and policy functions
+            ae.vc[ib, iy] = current_max
+            ae.policy[ib, iy] = pol_ind
+            ae.vf[ib, iy] = defval > current_max ? defval : current_max
+        end
+    end
+
+    nothing
+end
+
+function compute_prices!(ae::Arellano_Economy)
+    # Unpack parameters
+    β, γ, r, ρ, η, θ, ny, nB = _unpack(ae)
+
+    # Create default values with a matching size
+    vd_compat = repmat(ae.vd, nB)
+    default_states = vd_compat .> ae.vc
+
+    # Update default probabilities and prices
+    copy!(ae.defprob, default_states * ae.Π')
+    copy!(ae.q, (1 - ae.defprob) / (1 + r))
+
+    nothing
+end
+
+#=
+This performs value function iteration and stores all of the data inside
+the ArellanoEconomy type.
+=#
+function vfi!(ae::Arellano_Economy; tol=1e-8, maxit=10000)
```

Comment:
Could we re-use the vfi concept from all other models:

- Provide `bellman_operator`
- Call `compute_fixed_point` on that operator. 

I haven't read what you are doing here closely enough to see if it is vastly different from what we do for other models.